src/hotspot/share/opto/library_call.cpp

```diff
@@ -2042,11 +2046,21 @@ void LibraryCallKit::inline_math_mathExact(Node* math, Node *test) {
 template <typename OverflowOp>
 bool LibraryCallKit::inline_math_overflow(Node* arg1, Node* arg2) {
   typedef typename OverflowOp::MathOp MathOp;
+  bool use_builtin_throw = false;
+  if (builtin_throw_applies(Deoptimization::Reason_intrinsic).first) {
+    // If builtin_throw would work (notably, the throw is hot and we don't care about backtraces),
+    // instead of bailing out on intrinsic or potentially deopting, let's do that!
+    use_builtin_throw = true;
+  } else if (too_many_traps(Deoptimization::Reason_intrinsic)) {
```

Comment:
Why `too_many_traps(Deoptimization::Reason_intrinsic)` check is not enough here?